File: backend/api/schemas/broker.py
"""Broker algorithm configuration schemas."""

from __future__ import annotations

from datetime import datetime
from typing import List
from pydantic import BaseModel, Field


class StrategyParameter(BaseModel):
    """Strategy parameter configuration."""

    fieldName: str
    stringValue: str
    disable: str
    dataType: str = "string"
    description: str = ""


class StrategyConfig(BaseModel):
    """Strategy configuration for a broker."""

    name: str
    parameters: List[StrategyParameter]


class BrokerAlgorithmConfig(BaseModel):
    """Broker algorithm configuration."""

    broker: str
    strategies: List[StrategyConfig]


class BrokerAlgorithmStorage(BaseModel):
    """Storage wrapper for broker algorithm data."""

    version: str = "1.0"
    lastUpdated: str = Field(default_factory=lambda: datetime.now().isoformat())
    configs: List[BrokerAlgorithmConfig]

File: ExecutionView/module/types/broker.ts
/**
 * Broker/strategy configuration types.
 *
 * P3-SRP: Extracted from types/index.ts.
 */

export interface BrokerStrategyField {
  fieldName: string;
  disable: string;
  stringValue: string;
}

export interface BrokerStrategiesResponse {
  broker: string;
  assetClass: string;
  strategies: string[];
}

export interface BrokerStrategyInfoResponse {
  broker: string;
  strategy: string;
  assetClass: string;
  fields: BrokerStrategyField[];
}

export interface StrategyParameter {
  fieldName: string;
  stringValue: string;
  disable: string;
  order?: number;
  dataType: 'string' | 'number' | 'boolean';
  description: string;
}

export interface StrategyConfig {
  name: string;
  parameters: StrategyParameter[];
}

export interface BrokerAlgorithmConfig {
  broker: string;
  assetClass?: string;
  strategies: StrategyConfig[];
}
